AutoFlow_OmicsDataHandling/INCA_script_generator.py

```diff
@@ -0,0 +1,1232 @@
+    """
+    Process data that is stored in strings of lists etc in the files
+
+    Parameters:
+        string: Processes the data in cells in the dataframes.
+            The info is either bordered by curly or double square
+            brackets.
+        type_of_replacement: Define the type of surrounding brackets
+
+    Returns:
+        string: returns data in lists without bordering brackets
+
+    """
+    if type_of_replacement == "Curly":
+        string = string.strip("}{").split(",")
+    elif type_of_replacement == "Double_square":
+        string = string[1:-1]
+        string = ast.literal_eval(string)
+    return string
+
+
+def initiate_MATLAB_script():
+    """
+    Starts writing the MATLAB script
+
+    Returns:
+        mat_script: Initialized MATLAB script
+
+    """
+    mat_script = "clear functions\n\n"
+
+    return mat_script
+
+
+def reaction_mapping(
+    atomMapping_molecules_ids,
+    model_molecules_ids,
+    atomMapping_molecules_stoichiometry,
+    atomMapping_molecules_elements,
+    atomMapping_molecules_mapping,
+    model_molecules_stoichiometry,
+    reaction_type="reactant",
+):
+    """
+    Provides the carbon mapping for metabolites in a model.
+    Called within "add_reactions_to_script()"
+
+    Parameters:
+        atomMapping_molecules_ids: processed atomMappingReactions_data_I
+            reactants_ids_tracked or products_ids_tracked
+        model_molecules_ids: processed modelReaction_data_I
+            reactants_ids or products_ids
+        atomMapping_molecules_stoichiometry: processed
+            atomMappingReactions_data_I
+            reactants_stoichiometry_tracked or productss_stoichiometry_tracked
+        atomMapping_molecules_elements: processed atomMappingReactions_data_I
+            reactants_elements_tracked or products_elements_tracked
+        atomMapping_molecules_mapping: processed atomMappingReactions_data_I
+            reactants_mapping or products_mapping
+        model_molecules_stoichiometry: processed modelReaction_data_I
+            reactants_stoichiometry or products_stoichiometry
+        reaction_type: reactant or product
+
+    Returns:
+        rxn_equation: Reaction equation for the defined reaction
+
+    """
+    rxn_equation = ""
+    if reaction_type == "product":
+        rxn_equation += " -> "
+    # This had to be added because some metabolites appear more than once
+    # in one reaction with different mappings
+    seen_molecule = {}
+    dupes_molecule = []
+    if not len(atomMapping_molecules_ids) == len(
+        set(atomMapping_molecules_ids)
+    ):
+        for x in atomMapping_molecules_ids:
+            if x not in seen_molecule:
+                seen_molecule[x] = 1
+            else:
+                if seen_molecule[x] == 1:
+                    dupes_molecule.append(x)
+                seen_molecule[x] += 1
+
+    # Iterate through the metabolites involved in the reactions
```

Comment:
Thanks for adding in these detailed comments.  This helps a lot in better understanding the code since the data structures are a bit hard to understand without actually running the code.